homeassistant/components/sensor/worldtidesinfo.py

```diff
@@ -0,0 +1,135 @@
+"""
+Support for retrieving tide levels from worldtides.info.
+
+configuration.yaml
+
+sensor:
+  - platform: worldtidesinfo
+    api_key: "YOUR API KEY"
+    latitude: "LATITUDE OF BEACH"
+    longitude: "LONGITUDE OF BEACH"
+"""
+import logging
+import time
+from datetime import timedelta
+
+import homeassistant.helpers.config_validation as cv
+import requests
+import voluptuous as vol
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=3600)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_LATITUDE): cv.string,
+    vol.Required(CONF_LONGITUDE): cv.string,
```

Comment:
There is a [validator](https://home-assistant.io/developers/development_validation/) for longitude available.